pkg/server/endpoints/endpoints.go

```diff
@@ -163,8 +164,10 @@ func (e *endpoints) runGRPCServer(ctx context.Context, server *grpc.Server) erro
 		return err
 	case <-ctx.Done():
 		e.c.Log.Info("Stopping gRPC server")
+		l.Close()
```

Comment:
There is a `defer l.Close()` statement above. Did you run into something that required it be done before calling `server.Stop()`?

Reply:
The defer close is important so that we close the listener in case of an error (say with the grpc hook), but yeah, during my testing the server wouldn't shutdown gracefully (server.Stop() would  not return) unless I closed the listener first.